### PR TITLE
add Code Climate static analysis

### PR DESCRIPTION
**Pull request by afeld**

### CODE REVIEW

**Comment by afeld** [comment]
Ready! I also set up the pull request integration, but not sure why it isn't appearing here.

A couple of engines that I _wanted_ to enable but couldn't:
- `markdownlint`: [the rules aren't configurable yet](https://github.com/jpignata/codeclimate-markdownlint/issues/6), and they enforce [line length](https://github.com/mivok/markdownlint/blob/v0.2.1/docs/RULES.md#md013---line-length) default, which I'm pretty strongly opposed to.
- `nodesecurity`: it complains that there isn't a `package.json` at the top level of the repository. I kinda like keeping tests/dependencies scoped to each task directory, so probably just going to skip this engine.

**Comment by afeld** [comment]
...but was able to verify locally.

```
$ docker run \
  --interactive --tty --rm \
  --env CODECLIMATE_CODE="$PWD" \
  --volume "$PWD":/code \
  --volume /var/run/docker.sock:/var/run/docker.sock \
  --volume /tmp/cc:/tmp/cc \
  codeclimate/codeclimate analyze
Starting analysis
Running duplication: Done!
Running eslint: Done!
Running fixme: Done!
Running shellcheck: Done!

Analysis complete! Found 0 issues.
```

File: .eslintrc [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+  no-dupe-class-members: 0
+  no-this-before-super: 0
+  no-var: 0
+  object-shorthand: 0
+  prefer-arrow-callback: 0
+  prefer-const: 0
+  prefer-reflect: 0
+  prefer-spread: 0
+  prefer-template: 0
+  require-yield: 0
```
Comment (afeld) [inline .eslintrc]: Note: This file was generated by Code Climate. I didn't spend any time customizing it, but imagine it could be simplified.

**Comment by DavidEBest** [comment]
This looks good, but shouldn't we add a ruby engine or two?

**Comment by afeld** [comment]
We don't have any Ruby in there yet, but I suppose we will soon enough. Added!

**Comment by mrb** [comment]
Can I just say? :clap:

**Pull request merged**